### PR TITLE
Add detail to trajectory tooltips (#35)

### PR DESCRIPTION
**Pull request by blms**
## In this PR

- Per #35:
  - In the trajectories visualization, when a user hovers over a place marker, display a list of waypoints on that place for the selected authors and time range
- Improve ESLint autofix behavior for destructured function arguments

## Questions

- Should we pull out the following snippet and make it reusable? It's copy pasted from `IntersectionDetails` right now, not DRY! Just wasn't sure where to put it since it's a combo of reducing the search results and filtering the waypoints.
  https://github.com/performant-software/itsb/blob/a1b8effd804e33e11b2b838a3b6c996cd06802f7/src/components/ITSBTooltip/ITSBTooltip.js#L36-L40
- I tried to keep the tooltips as minimal as possible, and one way of doing this is that I only show "unlabeled" waypoints (i.e. only author/year but no `relation.label`) when there are _no_ waypoints with a label on that place. Do you think that solution is ok? Is it misleading to only _sometimes_ surface that data?
  - Here's basically why I made this change: [displaying all](https://user-images.githubusercontent.com/7234006/205166468-bdc096fb-f894-4fd9-be0b-5ce780d3c6e5.png) vs [skipping "unlabeled"](https://user-images.githubusercontent.com/7234006/205166406-45c13092-3294-41e1-b7e0-38e51206d6ad.png). But maybe having the full time range selected is an unusual use case?
- I think this is the first time I'm really interacting with the search and graph providers in my code; did I make any mistakes? 😅 
- I wonder if it's possible to keep toolips from overflowing the viewport… but probably out of scope for this, unless we just want to reduce the `max-width`.

### CODE REVIEW

**Comment by rsimon** [comment]
> Should we pull out the following snippet and make it reusable?

I'll think about that... I'm actually using another variation of the same functionality now in the TrajectoriesLayer. (Although I'm pulling out all waypoint from the graph first, then filter by authors in the search results.) But definitely agree: we should keep this DRY.

**Comment by rsimon** [comment]
> I tried to keep the tooltips as minimal as possible

It's really tricky to find a good solution for this - without adding an additional component (or resorting to a third column again - which [Alex specifically did not prefer](https://github.com/performant-software/itsb/issues/35#issuecomment-1322677599). 

A possible tweak, perhaps, could be to add an extra list item per author, with a label like "2 additional visits 1952-180"

**Comment by rsimon** [comment]
> I wonder if it's possible to keep toolips from overflowing the viewport…

Yikes - c'mon... I would have expected DeckGL to handle this. Well, likewise: since the DeckGL seems to force the tooltip position down->left from the map marker, I think we only have the choice between accepting the default behavior, or building our own popup.

FWIW: I built a custom popup implementation for the MapLibre (=non-DeckGL) map type in Peripleo, which gets checks position and then re-positions to stay in the visible viewport. I'll take a look if I can tweak that to make it work on the DeckGL map type, too. (Let's merge first, though!)

File: src/components/ITSBTooltip/ITSBTooltip.js [hunk 1]
```diff
@@ -1,3 +1,66 @@
...
+ * @returns {string} Date formatted MM/DD/YYYY.
+ */
+function fmt(date) {
+  const split = date?.split('-');
+  if (split?.length) {
+    split.push(split.shift());
+  }
+  return split?.join('/') || '?';
+}
+
```
Comment (rsimon) [inline src/components/ITSBTooltip/ITSBTooltip.js:17]: I think you can just re-use `formatInterval(waypoint)` from `components/ITSBStore/utils/waypointUtils.js`.

**Review by rsimon — APPROVED** [review]

**Comment by blms** [comment]
> Update: I think a suitable place to put this could be the `itineraryUtils` in `components/ITSBStore/utils`. Feels closest to processing a list of itineraries and deriving a certain result from them.

Good call, done!

> A possible tweak, perhaps, could be to add an extra list item per author, with a label like "2 additional visits 1952-180"

Done, I really like this solution to avoid hiding any data.

I also noticed that even with hiding them, showing _just_ W.E.B. Du Bois and hovering Atlanta overflows the page. But I think there's no avoiding that with the tooltips, so we will just have to let Alex decide how he feels about it. Maybe it just indicates to users that they should narrow their date range 😆 

> I think you can just re-use `formatInterval(waypoint)` from `components/ITSBStore/utils/waypointUtils.js`.

Great catch, I'd missed that util.

**Review by rsimon — APPROVED** [review]
Excellent thanks!

**Pull request merged**